### PR TITLE
Version 1 : Few changes for grid items and backlog.

### PR DESCRIPTION
**Pull request by bsmaheshkumar**
screen shot 2018-01-01 at 3 02 01 pm

1. Added few filters
2. Converted Recommended Date from String to Date. (Not able to display
Time in dd—MMM-YYYY) formate, maybe the right way to display is using
Pipes…
3. Added Days (duration of recommendation), is the right design to use
Pipe’s here? Even the Data type is number, the sorting is not working
4. Updated Product backlog with few new ideas.

### CODE REVIEW

File: wealth-management-client/src/app/modal/stock.ts [hunk 1]
```diff
@@ -1,25 +1,35 @@
...
-    this.targetPrice = data.targetPrice;
+    this.investPrice = data.buyPrice;
+
     this.rcomPrice = data.rcomPrice;
-    this.rcomDate = data.rcomDate;
+    this.targetPrice = data.targetPrice;
+    this.rtargetPercentage = (((data.targetPrice - data.rcomPrice) / data.rcomPrice) * 100).toFixed(2);
+
+    this.rcomDate = new Date(data.rcomDate);
+    this.days = Math.abs((firstDate.valueOf() - secondDate.valueOf()) / (24 * 60 * 60 * 1000)).toFixed(0);
```
Comment (bsmaheshkumar) [inline wealth-management-client/src/app/modal/stock.ts]: @coderhindu  is the right design to use pipes?

File: wealth-management-client/src/app/reports/recommendations/recommendation-grid.component.html [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
-  <p-column field="script" header="Script" sortable="true"></p-column>
-  <p-column field="name" header="Name" sortable="true"></p-column>
+  <p-column field="script" header="Script" sortable="true" filter="true"></p-column>
+  <!-- p-column field="name" header="Name" sortable="true"></p-column -->
   <p-column field="rcomPrice" header="Recommended Price" sortable="true"></p-column>
-  <p-column field="buyPrice" header="Buy Price" sortable="true"></p-column>
+  <!-- p-column field="buyPrice" header="Buy Price" sortable="true"></p-column -->
   <p-column field="targetPrice" header="Recommended Price" sortable="true"></p-column>
+  <p-column field="rtargetPercentage" header="Recommended Target" sortable="true"></p-column>
+  <p-column field="days" header="Number of Days" sortable="true"></p-column>
```
Comment (bsmaheshkumar) [inline wealth-management-client/src/app/reports/recommendations/recommendation-grid.component.html]: Sort is not working...

File: wealth-management-client/src/app/modal/stock.ts (hunk 1 above)
Comment (coderhindu) [reply wealth-management-client/src/app/modal/stock.ts]: Yes, view changes can be handled in pipes. No offence doing here.

File: wealth-management-client/src/app/reports/recommendations/recommendation-grid.component.html (hunk 2 above)
Comment (coderhindu) [reply wealth-management-client/src/app/reports/recommendations/recommendation-grid.component.html]: Sort was working for me. Is it not working after adding filter?

**Comment by neerajkujha** [comment]
The header label should be **scrip, suggested price, target price, target %age**. Also i am not able to compile. getting following error
ERROR in src/app/modal/stock.ts(26,5): error TS2322: Type 'string' is not assignable to type 'number'.
src/app/modal/stock.ts(29,5): error TS2322: Type 'string' is not assignable to type 'number'.

**Pull request merged**